connector/src/main/java/org/jboss/as/connector/subsystems/datasources/GetDataSourceClassInfoOperationHandler.java

```diff
@@ -0,0 +1,235 @@
+     *
+     * @see Injection.findMethod()
+     * @param clz the class
+     * @return whether it is understandable
+     */
+    private static boolean isTypeMatched(Class<?> clz) {
+        if (clz.equals(String.class)) {
+            return true;
+        } else if (clz.equals(byte.class) || clz.equals(Byte.class)) {
+            return true;
+        } else if (clz.equals(short.class) || clz.equals(Short.class)) {
+            return true;
+        } else if (clz.equals(int.class) || clz.equals(Integer.class)) {
+            return true;
+        } else if (clz.equals(long.class) || clz.equals(Long.class)) {
+            return true;
+        } else if (clz.equals(float.class) || clz.equals(Float.class)) {
+            return true;
+        } else if (clz.equals(double.class) || clz.equals(Double.class)) {
+            return true;
+        } else if (clz.equals(boolean.class) || clz.equals(Boolean.class)) {
+            return true;
+        } else if (clz.equals(char.class) || clz.equals(Character.class)) {
+            return true;
+        } else if (clz.equals(InetAddress.class)) {
+            return true;
+        } else if (clz.equals(Class.class)) {
+            return true;
+        } else if (clz.equals(Properties.class)) {
+            return true;
+        }
+        return false;
+    }
+
+    private static String deCapitalize(String str) {
+        if (str.length() == 1) {
+            return str.toLowerCase(Locale.US);
+        }
+        if (str.equals(str.toUpperCase(Locale.US))) { // all uppercase, just return
+            return str;
+        }
+        return str.substring(0, 1).toLowerCase(Locale.US) + str.substring(1);
+    }
+
+    private static ClassLoader getClassLoader(final Class<?> clz) {
+        if (System.getSecurityManager() == null)
+            return clz.getClassLoader();
+        return AccessController.doPrivileged(new PrivilegedAction<ClassLoader>() {
+            public ClassLoader run() {
+                return clz.getClassLoader();
+            }
+        });
+    }
+
+    /**
+     * Get the declared methods
+     *
+     * @param c The class
+     * @return The methods
+     */
+    private static Method[] getDeclaredMethods(final Class<?> c) {
+        if (System.getSecurityManager() == null)
+            return c.getDeclaredMethods();
+
+        return AccessController.doPrivileged(new PrivilegedAction<Method[]>() {
+            public Method[] run() {
+                return c.getDeclaredMethods();
+            }
+        });
+    }
+
+    private static List<Method> allPublicSetterMethods(Class<?> clz) {
+        List<Method> hits = new ArrayList<>();
+        while (!clz.equals(Object.class)) {
+            Method[] methods = getDeclaredMethods(clz);
+            for (int i = 0; i < methods.length; i++) {
+                final Method method = methods[i];
+                if (Modifier.isPublic(method.getModifiers())
+                        && method.getName().startsWith("set")
```

Comment:
You also need to check if it is static and that the length>3 (so you don't pick up a method called "set")